### PR TITLE
Fix deleting in grades system by using .update for Firestore.

### LINKED ISSUE #1505: Deleting grade doesn't work
https://github.com/SharezoneApp/sharezone-app/assets/24459435/b6d47841-9d91-4dd8-a2be-08b60142e333

### PR DESCRIPTION
**Pull request by Jonas-Sander**
Fixes #1505

### CODE REVIEW

File: app/lib/grades/grades_service/src/grades_repository.dart [hunk 1]
```diff
@@ -97,26 +97,27 @@ class FirestoreGradesStateRepository extends GradesStateRepository {
...
+    // Adding the top level createdOn should've created the document if it
+    // didn't exist yet. Thus we can use update instead of set here.
+    gradesDocumentRef.update(data);
   }
 
-  Future<void> tryAddTopLevelCreatedOn() async {
-    final snap =
-        await gradesDocumentRef.get(const GetOptions(source: Source.cache));
+  Future<void> maybeAddTopLevelCreatedOn() async {
+    final snap = await gradesDocumentRef.get();
```
Comment (nilsreichardt) [inline app/lib/grades/grades_service/src/grades_repository.dart:118]: Will this break in offline mode?

**Review by nilsreichardt — APPROVED** [review]
LGTM

File: app/lib/grades/grades_service/src/grades_repository.dart (hunk 1 above)
Comment (nilsreichardt) [reply app/lib/grades/grades_service/src/grades_repository.dart:118]: I started app preview, with that it's easy to test it

**Comment by Sharezone-Admin** [comment]
⬇️ Builds generated by [Codemagic](https://codemagic.io) for commit 0746c13b5612bc3007eb5c894e47ae93c6ecbc17 ⬇️

_Note_: Only Sharezone team members are able to install the iOS app.

| iOS | | Android |
|:-:|:-:|:-:|
| iOS QR code <br /> [Download link](https://api.codemagic.io/artifacts/.eJwVwUtygjAAANC7dJ8ZCgmBhYugaAOCfJTfJgMIgihgpAI5fafvfd3IP-NpvboQrPAjfU9UItajyrxPzg7BjELs8yUMwokgqHRmpVj45qnRXl1gZAeyV4TtgW6HuKLtpXnrAE3PIamYtejRfQJ1nS2_45pTouVn6Z40xXhFnh93vZLA7tNA-V09HobsYlzeLpyoVBUaA3EpdHK17-QnH7Xp5Ch9Z-5OhgjhMIpjjcb9hZqdxVJ2FinUeenEUWb5Jm40wwU1s5ld9b0x5O8y4b5udu55mY9eHqMiVcBpG68YGIXrUF7vRSYH6Xr0SolKPIWtjnYct8Ms5sl5Rf5m8_UHfFVg4w.6MnZVg9nclzH4dmQIhbrSSKsCqA) | | Android QR code <br /> [Download link](https://api.codemagic.io/artifacts/.eJwVwUmSgjAAAMC_eKeKjIPAwQMSCDEyIHu4UKhBVDQBZX29Nd2rq_Fv99y3j5CIIiB8SER6NBYTDG6gzaIVh1vm9Sm7lelaPfvD68KPMbWg3Cof3ZhLhVmIhsGPok8kncoMi4bNfK7s-9OGZvjX6DlHoivcVo3qfilqgQ16QxtPXYo3Td6MSYcYWKCKaTeBkeXA03qF4lbXIiXT8gviRwIB8GwsO1XzSEzXPw9QmAGClix3Y030TfYgun9gUjTgz3U_Lt7g7joyavv6ebaVfC14dTKh5ElJAmJOUVmVBoaiQSBT7y_gq--Nc7-Upy6e-Ihi5zdxnNyK8Nwb2-3qCyTEYfg.BiaZkMFP2Zc7D5QGQKNj4leNDH4) |

<sub>(expires Wed, 22 May 2024 18:14:42 GMT)<br/>Codemagic build: [6626a5912c114a073a1cfef3](https://codemagic.io/app/629c82ea463af7ff553fc7a5/build/6626a5912c114a073a1cfef3); comment generated with [Codemagic App Preview](https://github.com/nilsreichardt/codemagic-app-preview)</sub>

**Pull request merged**